src/__tests__/poller.test.ts

```diff
@@ -0,0 +1,75 @@
+import Poller from '../poller'
+
+jest.useFakeTimers()
+
+describe('Poller', () => {
+  let fetchFlagsFn: jest.Mock
+  let configurations: object
+  let pollInterval: number
+  let maxAttempts: number
+
+  beforeEach(() => {
+    configurations = {}
+    fetchFlagsFn = jest.fn()
+    pollInterval = 60000
+    maxAttempts = 5
+  })
+
+  it('should not start polling if it is already polling', () => {
+    const poller = new Poller(fetchFlagsFn, configurations, pollInterval, maxAttempts)
+    const logSpy = jest.spyOn(poller as any, 'logDebug')
+
+    poller.start()
+    poller.start()
+
+    expect(logSpy).toHaveBeenCalledWith(
+      'Polling was requested but is already running - only one poller can run at a time.'
+    )
+  })
+
+  it('should retry fetching if there is an error', async () => {
+    fetchFlagsFn.mockResolvedValue(new Error('Fetch Error'))
+    const poller = new Poller(fetchFlagsFn, configurations, pollInterval, maxAttempts)
+    const logSpy = jest.spyOn(poller as any, 'logDebug')
+
+    poller.start()
+    jest.advanceTimersByTime(maxAttempts * pollInterval)
+    await Promise.resolve()
+    expect(fetchFlagsFn).toHaveBeenCalledTimes(2)
+    expect(logSpy).toHaveBeenCalledWith('Error when polling for flag updates', expect.any(Error))
```

Comment:
In tests, try to visually distinguish between actions and assertions. For someone to understand this test, they'd need to read all lines, but a few line breaks will allow quick scanning.
```suggestion
    poller.start()
    
    jest.advanceTimersByTime(maxAttempts * pollInterval)
    await Promise.resolve()
    
    expect(fetchFlagsFn).toHaveBeenCalledTimes(2)
    expect(logSpy).toHaveBeenCalledWith('Error when polling for flag updates', expect.any(Error))
```